xcp/cmd.py

```diff
@@ -24,37 +24,63 @@
 """Command processing"""
 
 import subprocess
+import sys
 
 from xcp import logger
 from xcp.compat import open_defaults_for_utf8_text
 
+def _encode_command_to_bytes(command):
+    # When the locale not an UTF-8 locale, Python3.6 Popen can't deal with ord() >= 128
+    # when the command contains strings, not bytes. Therefore, convert any strings to bytes:
+    if sys.version_info >= (3, 0):
+        if isinstance(command, str):
+            command = command.encode()
+        elif isinstance(command, list):
```

Comment:
what if it fails both type checks?

Reply:
@psafont Thanks, good catch!

* Updated the function to support all types of `iterable`s and sequences supported by `Popen()`
* The additional test case tests and covers the added `raise TypeError()`